src/lint.ts

```diff
@@ -44,6 +44,13 @@ export interface LintResult {
     output: string;
 }
 
+export interface ILinterOptionsRaw {
```

Comment:
There's a TypeScript issue filed somewhere for letting us do something like `ILinterOptions?`, where every field in `ILinterOptions` would be marked optional (so exactly this).

My problem with re-using one interface for this is that after this initial parsing, the fields *aren't* optional. The semantic intent changes from input option settings to stored options data.